src/System/FSNotify.hs

```diff
@@ -207,8 +208,8 @@ threadChan listenFn (WatchManager db listener cleanupVar) path actPred action =
         , stopListener >> cleanThisUp
         )
 
-readEvents :: EventChannel -> Action -> IO ()
-readEvents chan action = forever $ do
+readEvents :: WatchConfig -> EventChannel -> Action -> IO ()
+readEvents (WatchConfig _ _ _ True) chan action = forever $ do
```

Comment:
Could you change this to be `(WatchConfig {confThreadPerEvent=True})` and the same for `False`?

Reply:
`hlint` also suggested to remove the `()` around record pattern. So I did that also